src/components/file-uploader/FileUploader.tsx

```diff
@@ -0,0 +1,96 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import Button from '@mui/material/Button'
+import IconButton from '@mui/material/IconButton'
+import ListItem from '@mui/material/ListItem'
+import List from '@mui/material/List'
+import CloudUploadIcon from '@mui/icons-material/CloudUpload'
+import CloseIcon from '@mui/icons-material/Close'
+import { SxProps } from '@mui/material'
+
+import useUpload from '~/hooks/use-upload'
+
+import { styles } from '~/components/file-uploader/FileUploader.styles'
+import { AddDocuments, ComponentEnum, Emitter, SizeEnum } from '~/types'
+import { spliceSx } from '~/utils/helper-functions'
+
+interface FileUploaderProps {
+  buttonText: string
+  emitter: ({ files, error }: Emitter) => void
+  initialState: File[]
+  initialError: string
+  validationData: AddDocuments
+  isImages?: boolean
+  sx: {
+    root?: SxProps
+    button?: SxProps
+  }
+}
+
+const FileUploader: FC<FileUploaderProps> = ({
+  buttonText,
+  emitter,
+  initialState = [],
+  initialError = '',
+  validationData,
+  isImages = false,
+  sx = {}
```

Comment:
here you set default value to sx, so I think you should make it not required

Reply:
If I delete this default value, I will receive an error. So I don't delete it.

Reply:
No, I mean, that as you set default value to sx, you maybe need to make it not required: "sx?"

Reply:
Added "sx?"